buildSrc/src/main/kotlin/TargetsConfig.kt

```diff
@@ -52,31 +53,47 @@ fun Project.configureTargets() {
 
         if (hasPosix || hasLinux || hasDarwin || hasWindows) extra.set("hasNative", true)
 
-        sourceSets {
-            if (hasJsAndWasmShared) {
-                val commonMain by getting {}
-                val jsAndWasmSharedMain by creating {
-                    dependsOn(commonMain)
-                }
-                val commonTest by getting {}
-                val jsAndWasmSharedTest by creating {
-                    dependsOn(commonTest)
+        @OptIn(ExperimentalKotlinGradlePluginApi::class)
+        val ktorTargetsHierarchy = KotlinHierarchyTemplate {
+            // extract nix group so it can be reused later as `group("nix")`
+            group("nix") {
```

Comment:
just curious[2]: is there a reason to declare it top-level and not under `common`? Does it make any difference?

Reply:
it can be moved under common. however then it might look not entirely correct as I don't want nix to be dependsOn common, but only on posix and jvmAndNix (but apply target hierarchy algorithm is capable to solve this problem as in this case it will not add direct dependsOn edge).

again I wasn't doing that with much of a through thinking of where exactly to place this group.

Reply:
> but apply target hierarchy algorithm is capable to solve this problem as in this case it will not add direct dependsOn edge

or maybe not, I need to double check

Reply:
> or maybe not, I need to double check
checked. it is alright. here is the source set structure I get via this script:

```
tasks.register("printKMPSourceSetStructure") {
    doFirst {
        println("digraph {")
        kotlin.sourceSets.forEach { sourceSet ->
            sourceSet.dependsOn.forEach { dependsOn ->
                println("  \"${sourceSet.name}\" -> \"${dependsOn.name}\"")
            }
        }
        println("}")
    }
}
```

![image](https://github.com/ktorio/ktor/assets/3123432/cfeca71c-af08-441e-ad9a-a1a38937b0f6)

Reply:
cool, thanks for explaining and checking :)